lib/fastlane/plugin/wpmreleasetoolkit/actions/common/get_prs_list_action.rb

```diff
@@ -6,38 +6,23 @@ module Actions
     class GetPrsListAction < Action
       def self.run(params)
         repository = params[:repository]
-        start_tag = params[:start_tag]
-        end_tag = params[:end_tag]
-        report_path = params[:report_path]
+        report_path = File.expand_path(params[:report_path])
+        milestone = params[:milestone]
 
         # Get commit list
-        commit_list = sh("git log --pretty=oneline #{start_tag}..#{end_tag}")
+        pr_list = Fastlane::Helper::GithubHelper.get_prs_for_milestone(repository, milestone)
 
-        # Extract PRs
-        pr_list = []
-        commit_list.split("\n").each do |commit|
-          if commit.include?('Merge pull request #')
-            # PR found, so extract PR number
-            pr_list.push(commit.partition('#').last.split(' ')[0])
-          end
-        end
-
-        # Get infos from GitHub and put into the target file
-        client = Fastlane::Helper::GithubHelper.github_client()
         File.open(report_path, 'w') do |file|
-          pr_list.each do |pr_number|
-            begin
-              data = client.pull_request(repository, pr_number.to_i)
-              file.puts("##{data[:number]}: #{data[:title]} @#{data[:user][:login]} #{data[:html_url]}")
-            rescue
-              UI.message("Could not find a PR with number #{pr_number.to_i}. Usually this is due to a bad reference in a commit message, but you probably want to check.")
-            end
+          pr_list.each do |data|
+            file.puts("##{data[:number]}: #{data[:title]} @#{data[:user][:login]} #{data[:html_url]}")
```

Comment:
Just in case you want to get fancy, you could replace this whole `File.open(…) do … end` block with this:
```ruby
lines = pr_list.map { |pr| "##{pr[:number]}: #{pr[:title]} @#{pr[:user][:login]} #{pr[:html_url]}" }
File.write(report_path, lines.join("\n"))
```

_(feel free to ignore, that's just me being pedantic 😇 )_